### PR TITLE
Add ci config for publishing docker image

### LINKED ISSUE #349: Build and push docker image during CI/CD
Consider to change the CI/CD configuration as so:
  * for any commit to the `develop` branch the docker image must be built and pushed as [`poanetwork/tokenbridge-contracts`](https://hub.docker.com/repository/docker/poanetwork/tokenbridge-contracts) with the version `develop-XXXXXXXX` where `XXXXXXXX` is the first 8 hexadecimal symbols from the commit revision.
  * for any commit to the `master` branch the docker image must be built and pushed as [`poanetwork/tokenbridge-contracts`](https://hub.docker.com/repository/docker/poanetwork/tokenbridge-contracts) with the version `master-XXXXXXXX` where `XXXXXXXX` is the first 8 hexadecimal symbols from the commit revision.
  * as soon as a new release is tagged the docker image must be built and pushed as [`poanetwork/tokenbridge-contracts`](https://hub.docker.com/repository/docker/poanetwork/tokenbridge-contracts) with the version matching the the release version. The same image must be pushed as `latest`.

### PR DESCRIPTION
**Pull request by k1rill-fedoseev**
This PR adds a GitHub Actions workflows for building and pushing docker images as described in #349.
Closes #349

### CODE REVIEW

File: .github/workflows/publish-docker-image-on-tags.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          repository: ${{ secrets.DOCKER_USERNAME }}/tokenbridge-contracts
```
Comment (akolotov) [inline .github/workflows/publish-docker-image-on-tags.yml:17]:
```suggestion
          repository: poanetwork/tokenbridge-contracts
```

File: .github/workflows/publish-docker-image-on-branch.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: evaluate-tag
+        run: echo "::set-output name=tag::${GITHUB_REF#refs/heads/}-${GITHUB_SHA::8}"
+      - uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          repository: ${{ secrets.DOCKER_USERNAME }}/tokenbridge-contracts
```
Comment (akolotov) [inline .github/workflows/publish-docker-image-on-branch.yml:20]:
```suggestion
          repository: poanetwork/tokenbridge-contracts
```

**Review by akolotov — CHANGES_REQUESTED** [review]

**Review by akolotov — COMMENTED** [review]
Will the images be built and published only in case if other jobs are successful?

**Comment by k1rill-fedoseev** [comment]
Initially, it was not so.
But now I have found a workaround how to place everything in a single workflow. So everything should work as expected.

**Review by akolotov — APPROVED** [review]
Ok. Let's try

**Pull request merged**